notebook/notebookapp.py

```diff
@@ -1671,7 +1671,7 @@ def server_info(self):
     def write_server_info_file(self):
         """Write the result of server_info() to the JSON file info_file."""
         try:
-            with open(self.info_file, 'w') as f:
+            with io.open(self.info_file, 'w') as f:
```

Comment:
Can you add `encoding='utf-8'` here? It's not the same problem, but it would be good to do it while we're looking at this code.